pkg/validate/validate.go

```diff
@@ -238,3 +238,16 @@ func UUID(val string) error {
 	_, err := uuid.Parse(val)
 	return err
 }
+
+// VCenter validates that a given string is either a valid IP or a hostname.
+func VCenter(v string) error {
```

Comment:
Let's not make this specific to vSphere.
```suggestion
// Host validates that a given string is a valid URI host.
func Host(v string) error {
```